compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala

```diff
@@ -167,6 +177,168 @@ case class WorkflowCompiler(extras: Option[Extras],
+      None
+    }
+
+    // Generate Markdown description section
+    val headerMd = s"\n## Task ${description.name}"
+    val details = description.details.collect {
+      case v: JsValue => v.asJsObject
+    }
+    val md = details match {
+      case Some(JsObject(details)) => {
+        val filesMd = details.get(Constants.FileDependencies) match {
+          case Some(JsArray(a)) =>
+            listMd("Files") + a.map(v => listMd2(v.toString)).mkString
+          case _ => ""
+        }
+        val networkDockerImageMd = details.get(Constants.NetworkDockerImage) match {
+          case Some(JsString(s)) => listMd("Network Docker image") + listMd2(s)
+          case _                 => ""
+        }
+        val dynamicDockerImageMd = details.get(Constants.DynamicDockerImage) match {
+          case Some(JsBoolean(b)) if b => listMd("Docker image determined at runtime")
+          case _                       => ""
+        }
+        val dxInstanceTypeMd = details.get(Constants.DxInstanceType) match {
+          case Some(JsString(s)) => listMd("Hard-coded instance type") + listMd2(s)
+          case _                 => ""
+        }
+        filesMd + networkDockerImageMd + dynamicDockerImageMd + dxInstanceTypeMd
+      }
+      case _ => ""
+    }
+    md match {
+      case s: String if !s.isEmpty => Some(headerMd + md)
+      case _                       => None
+    }
+  }
+
+  // Summarize dependencies that are not bundled for cloning
+  // with the workflow for adding to description in Markdown
+  private def summarizeReportableDependencies(
+      workflow: Workflow,
+      executableDict: Map[String, CompiledExecutable],
+      execTree: JsObject,
+      nativeAppDependencies: Vector[String],
+      fileDependencies: Vector[String],
+      dockerRegistryCredentialsUri: Option[String]
+  ): Option[String] = {
+    val headerMd =
+      "# This workflow requires access to the following dependencies that are not " +
+        "packaged with the workflow"
+
+    val appsMd = nativeAppDependencies.map(listMd2).mkString match {
+      case s: String if !s.isEmpty => listMd("Native apps") + s
+      case _                       => ""
+    }
+    val filesMd = fileDependencies.map(listMd2).mkString match {
+      case s: String if !s.isEmpty => listMd("Files") + s
+      case _                       => ""
+    }
+    val dockerRegistryCredentialsMd = dockerRegistryCredentialsUri match {
+      case Some(s) => listMd("Docker registry credentials file") + listMd2(s)
+      case _       => ""
+    }
+    val topLevelMd = appsMd + filesMd + dockerRegistryCredentialsMd
+
+    val executableNames = ExecutableTree.getExecutableNames(execTree)
+    val executables = executableNames.collect { n =>
+      executableDict.get(n)
+    }.flatten
+
+    val subWorkflowsMd = executables
+      .collect {
+        case CompiledExecutable(Workflow(_, _, _, _, _, _, _, _, _, _, _), dxExec, _, _) => dxExec
```

Comment:
use `workflow: DxWorkflow` instead

Reply:
Can you check the code for this part -- is that what you meant, or is there a better way?

Reply:
you can put `dxExec: DxWorkflow` in the first case statement so you don't have to do nested matches.